compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2667,7 +2667,10 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                 self.normalize_ty(ast_ty.span, array_ty)
             }
             hir::TyKind::Typeof(ref e) => {
-                let ty = tcx.type_of(tcx.hir().local_def_id(e.hir_id));
+                let ty_erased = tcx.type_of(tcx.hir().local_def_id(e.hir_id));
+                let ty = tcx.fold_regions(ty_erased, |r, _| {
+                    if r.is_erased() { tcx.lifetimes.re_static } else { r }
```

Comment:
Likely makes more sense to never create the ReErased in the first place...

Reply:
That would require changing the `type_of` query for anon consts -- not sure what that would break though. I guess anon consts _should_ be treating all those lifetimes as ReStatic like other consts though :thinking:

Reply:
@jackh726: Oh... sooo this calls the `typeck` query on the body of the `AnonConst`, so all non-late lifetimes are erased during writeback. Don't think I can avoid getting `ReErased` when I call `tcx.type_of`.